.github/workflows/docker.yml

```diff
@@ -8,37 +8,32 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    name: Build and publish
 
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
 
     - name: Login to DockerHub
-      uses: docker/login-action@v1
+      id: login
+      uses: docker/login-action@v3
+      continue-on-error: true
```

Comment:
What's the purpose of this? If this means if login was unsuccessful then what's the point of continuing with the build?

Reply:
That is exactly what it does, and the point was to allow me to test the workflow without a docker account linked to my repo.

Reply:
Thanks for confirming and explanation. I suggest removing it from the code that's going to be merged (or perhaps commenting it out with explanation). This may cause issues if we accidentally misconfig the credentials and the build shows green, it'll harder to figure out something is wrong with the build.

Reply:
Sure, I'll comment it out with a short explanation :)

Reply:
I assume the check in the push state still works as expected.

Thank you for making the change.